contracts/contracts/AllowList.sol

```diff
@@ -1,5 +1,5 @@
 //SPDX-License-Identifier: MIT
-pragma solidity ^0.8.0;
+pragma solidity ^0.8.24;
```

Comment:
I've explicitly defined shangai version in hardhat config, but contracts should explicitly use <0.8.24 (cancun activation)